core/ecschema-editing/src/Validation/ECRules.ts

```diff
@@ -328,7 +329,7 @@ export async function* incompatibleValueTypePropertyOverride(property: AnyProper
     if (!baseType || primitiveType === baseType)
       return;
 
-    return new Diagnostics.IncompatibleValueTypePropertyOverride(property, [property.class.fullName, property.name, baseClass.fullName, primitiveTypeToString(baseType), primitiveTypeToString(primitiveType!)]);
+    return new Diagnostics.IncompatibleValueTypePropertyOverride(property, [property.class.fullName, property.name, baseClass.fullName, primitiveTypeToString(baseType), primitiveTypeToString(expectDefined(primitiveType))]);
```

Comment:
This code should be rewritten so the call back isn't reaching out of its scope to get primitiveType.  It would be clearer if primitiveType was passed in as the `overrideType` of something like that indicating that this the type of the property that is trying to override a property from the base class.  It would also avoid the need for expectDefined here.

Reply:
I discovered that the requirement for `!` or `expectDefined` was caused by what I feel is a compiler bug. (The compiler for some reason thinks it might be undefined in `callback`, even though it's a `const` that has already been verified to be defined.) Introducing another `const` removed the need for either `!` or `expectDefined`, so I made that change. Note that this doesn't address your actual comment here, but it does provide what I feel is a better fix to get rid of the `!`, and your comment doesn't _seem_ to be in the scope of this PR.